### PR TITLE
Update datasets task tags to align tags with models

### PR DESCRIPTION
**Pull request by lhoestq**
**Requires https://github.com/huggingface/datasets/pull/4066 to be merged first**

Following https://github.com/huggingface/datasets/pull/4066 we need to update many dataset tags to use the new ones. This PR takes case of this and is quite big - feel free to review only certain tags if you don't want to spend too much time on it.

Note that the CI will never be green for this PR, because many dataset cards have missing tags or sections, and fixing them is out of scope of this PR (the CI on master will be green anyway)

### CODE REVIEW

**Comment by lhoestq** [comment]
Just updated the tags of vision datasets :)
We can figure out one for image datasets without labels like PASS - not sure how to name the task for this, maybe `image-fill-mask` (for consistency with language modeling for pretraining) / `masked-auto-encoding` (from ViT). Let's see that in another PR later

File: datasets/svhn/README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ task_categories:
 - image-classification
 - object-detection
 task_ids:
-- single-label-image-classification
+- object-detection-other-digit-detection
```
Comment (mariosasko) [inline datasets/svhn/README.md:21]: Would remove this and define task ids as `task_ids: []`

**Review by mariosasko — COMMENTED** [review]
IMO we should use `multi-class image-classification` as `task_id` for the updated vision datasets (beans, cats_vs_dogs, cifar10, cifar100) and not `image-classification-other ...`.

File: datasets/svhn/README.md (hunk 1 above)
Comment (lhoestq) [reply datasets/svhn/README.md]: done !

**Review by mariosasko — APPROVED** [review]
LGTM!

**Pull request merged**